vscode/open/command.py

```diff
@@ -106,18 +135,19 @@ def command(shell, args: list):
         arch = get_endpoint_architecture(DEFAULT_MACHINE)
         dtslogger.info(f"Target architecture automatically set to {arch}.")
 
-        # get the module configuration
-        # noinspection PyListCreation
-        module_configuration_args = []
-        # apply default module configuration
-        module_configuration_args.append(f"--net={parsed.network_mode}")
-        
-        # Set docker network mode TODO
+        # Set docker network mode
+        if parsed.network_mode is not None:
+            container_configuration.network_mode = parsed.network_mode
 
         if parsed.ros is not None:
-            # Add VEHICLE_NAME as an environment variable of the container
-            DEVCONTAINER_TEMPLATE["containerEnv"]["VEHICLE_NAME"] = parsed.ros
-        
+            if isinstance(container_configuration.environment, dict):
+                # It's a dictionary
+                container_configuration.environment['VEHICLE_NAME'] = parsed.ros
+            elif isinstance(container_configuration.environment, list):
+                # It's a list
+                container_configuration.environment = [env for env in container_configuration.environment if not env.startswith('VEHICLE_NAME=')]
+                container_configuration.environment.append(f'VEHICLE_NAME={parsed.ros}')
+
         # add default mount points
```

Comment:
Some properties (such as the `environment` one) in the `docker-compose.yml` file can be specified as either a dictionary of `{'key': 'value',}` pairs or as a list `['key=value',]`. This can be problematic to handle in particular when extending a container configuration with the  `__extend__` attribute in `containers.yaml`.

Reply:
Let's force it to be a dictionary, we don't support the list version.

Reply:
ok. Here we set the `VEHICLE_NAME` to be the same as the ros master hostname and `ROS_MASTER_URI` should be set by the entrypoint script from `dt-commons`, however I suspect that this is not getting executed or it is overwritten as those variables do not change in the docker container.

Reply:
Implemented here and in the `ContainerConfiguration` class.